pureskillgg_dsdk/adx/api.py

```diff
@@ -0,0 +1,61 @@
+from .dataset import AdxDataset
+from .writer_fs import AdxDatasetWriterFs
+from .writer_s3 import AdxDatasetWriterS3
+
+
+def get_adx_dataset_revisions(dataset_id, /, *, start_date=None, end_date=None):
+    client = AdxDataset(dataset_id=dataset_id, writer=None)
+    return client.get_revisions(start_date, end_date)
+
+
+def download_adx_dataset_revision(
+    root_path, dataset_id, revision_id=None, /, *, prefix=None
+):
+    writer = AdxDatasetWriterFs(root_path=root_path, prefix=prefix)
+    client = AdxDataset(dataset_id=dataset_id, writer=writer)
+
+    rev_id = revision_id
+    if revision_id is None:
+        rev = client.get_latest_revision()
+        if rev is None:
+            raise RuntimeError("No revisions in dataset")
+        rev_id = rev["Id"]
+
+    client.export_revision(rev_id)
+
+
+def export_single_adx_dataset_revision_to_s3(
+    bucket, dataset_id, revision_id=None, /, *, prefix=None
+):
+    writer = AdxDatasetWriterS3(bucket=bucket, prefix=prefix)
+    client = AdxDataset(dataset_id=dataset_id, writer=writer)
+
+    rev_id = revision_id
+    if revision_id is None:
+        rev = client.get_latest_revision()
+        if rev is None:
+            raise RuntimeError("No revisions in dataset")
+        rev_id = rev["Id"]
+
+    client.export_revision(rev_id)
```

Comment:
```suggestion
    if revision_id is None:
        rev = client.get_latest_revision()
        if rev is None:
            raise RuntimeError("No revisions in dataset")
        revision_id= rev["Id"]

    client.export_revision(revision_id)
```

the output of 
```
banana=None
if banana is None:
    banana=2
print(banana)
```
is 2, so the swap here is weird.